test/new-e2e/utils/e2e/client/agent_cmd_runner.go

```diff
@@ -8,34 +8,41 @@ package client
 import (
 	"errors"
 	"regexp"
+	"testing"
 	"time"
 
-	"github.com/DataDog/test-infra-definitions/components/os"
 	"github.com/cenkalti/backoff"
+	"github.com/stretchr/testify/require"
 )
 
-// AgentCommandRunner provides methods to run commands on the Agent.
+// agentRawCommandRunner is a low level interface to that provide a method to run command on the Agent.
+type agentRawCommandRunner interface {
+	ExecuteWithError(arguments []string) (string, error)
+}
+
+// AgentCommandRunner provides high level methods to run commands on the Agent.
 type AgentCommandRunner struct {
-	cmdRunner *vmClient
-	os        os.OS
+	t             *testing.T
+	commandRunner agentRawCommandRunner
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What about embedding the raw runner, so that you can call `agent.ExecuteWithError` instead ?

Reply:
I don't think the method should be exposed.